src/services/aws_kms/mod.rs

```diff
@@ -0,0 +1,385 @@
+//! # AWS KMS Service Module
+//!
+//! This module provides integration with AWS KMS for secure key management
+//! and cryptographic operations such as public key retrieval and message signing.
+//!
+//! Currently only EVM is supported.
+//!
+//! ## Features
+//!
+//! - Service account authentication using credential providers
+//! - Public key retrieval from KMS
+//! - Message signing via KMS
+//!
+//! ## Architecture
+//!
+//! ```text
+//! AwsKmsService (implements AwsKmsServiceTrait)
+//!   ├── Authentication (via AwsKmsClient)
+//!   ├── Public Key Retrieval (via AwsKmsClient)
+//!   └── Message Signing (via AwsKmsClient)
+//! ```
+//! is based on
+//! ```text
+//! AwsKmsClient (implements AwsKmsClientTrait)
+//!   ├── Authentication (via shared credentials)
+//!   ├── Public Key Retrieval in DER Encoding
+//!   └── Message Digest Signing in DER Encoding
+//! ```
+//! `AwsKmsClientTrait` is mocked with `mockall` for unit testing
+//! and injected into `AwsKmsService`
+//!
+
+use alloy::primitives::keccak256;
+use async_trait::async_trait;
+use aws_config::{meta::region::RegionProviderChain, BehaviorVersion, Region};
+use aws_sdk_kms::{
+    primitives::Blob,
+    types::{MessageType, SigningAlgorithmSpec},
+    Client,
+};
+use serde::Serialize;
+
+use crate::{
+    models::{Address, AwsKmsSignerConfig},
+    utils::{self, derive_ethereum_address_from_der, extract_public_key_from_der},
+};
+
+#[cfg(test)]
+use mockall::{automock, mock};
+
+#[derive(Clone, Debug, thiserror::Error, Serialize)]
+pub enum AwsKmsError {
+    #[error("KMS response parse error: {0}")]
+    ParseError(String),
+    #[error("KMS config error: {0}")]
+    ConfigError(String),
+    #[error("KMS get error: {0}")]
+    GetError(String),
+    #[error("KMS signing error: {0}")]
+    SignError(String),
+    #[error("KMS permissions error: {0}")]
+    PermissionError(String),
+    #[error("KMS public key error: {0}")]
+    RecoveryError(#[from] utils::Secp256k1Error),
+    #[error("Other error: {0}")]
+    Other(String),
+}
+
+pub type AwsKmsResult<T> = Result<T, AwsKmsError>;
+
+#[async_trait]
+#[cfg_attr(test, automock)]
+pub trait AwsKmsServiceTrait: Send + Sync {
+    /// Returns the EVM address derived from the configured public key.
+    async fn get_evm_address(&self) -> AwsKmsResult<Address>;
+    /// Signs a payload using the EVM signing scheme.
+    /// Pre-hashes the message with keccak-256.
+    async fn sign_payload_evm(&self, payload: &[u8]) -> AwsKmsResult<Vec<u8>>;
+}
```

Comment:
Let's change the name of this trait. 
1. No need to name it `*Trait`, it is clear enough that this is a trait already.
2. It does not represent that this trait brings you EVM signature functionality.

Maybe something like `AwsKmsEVMService`?

It also applies to the next trait.

Reply:
Sure thing. I followed the convention from other services such as `GoogleCloudKmsServiceTrait` and `GoogleCloudKmsService`, but I am happy to rename